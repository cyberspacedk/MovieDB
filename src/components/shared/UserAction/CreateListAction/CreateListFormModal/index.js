import CreateListFormModal from './container';

export default CreateListFormModal;
