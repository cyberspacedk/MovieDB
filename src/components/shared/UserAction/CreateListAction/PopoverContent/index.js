import PopoverContent from './component';

export default PopoverContent;
