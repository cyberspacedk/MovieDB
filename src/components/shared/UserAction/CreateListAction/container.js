import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getCreatedListRequest,
  addMovieToListRequest,
} from '../../../../store/myLists/actions';
import { getMyList } from '../../../../store/myLists/selectors';
import CreateListAction from './component';

class CreateListActionContainer extends Component {
  state = {
    visiblePop: false,
    visibleMod: false,
  };

  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.getCreatedListRequest();
  }

  showPopover = visible => this.setState({ visiblePop: visible });

  hidePopover = () => this.setState({ visiblePop: false });

  showModal = () => this.setState({ visibleMod: true });

  hideModal = () => this.setState({ visibleMod: false });

  addMovieToList = (listId, movieId) => {
    this.props.addMovieToListRequest(listId, movieId);
    this.hidePopover();
  };

  render() {
    const { visiblePop, visibleMod } = this.state;

    return (
      <CreateListAction
        visiblePop={visiblePop}
        visibleMod={visibleMod}
        showPopover={this.showPopover}
        hidePopover={this.hidePopover}
        showModal={this.showModal}
        hideModal={this.hideModal}
        addMovieToList={this.addMovieToList}
        {...this.props}
      />
    );
  }
}

CreateListActionContainer.propTypes = {
  getCreatedListRequest: PropTypes.func.isRequired,
  addMovieToListRequest: PropTypes.func.isRequired,
  myLists: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mstp = store => ({
  myLists: getMyList(store),
});

const mdtp = {
  getCreatedListRequest,
  addMovieToListRequest,
};

export default connect(
  mstp,
  mdtp,
)(CreateListActionContainer);
