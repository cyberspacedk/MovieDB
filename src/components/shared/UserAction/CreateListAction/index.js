import CreateListAction from './container';

export default CreateListAction;
