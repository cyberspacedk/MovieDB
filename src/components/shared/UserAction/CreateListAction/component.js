import React from 'react';
import PropTypes from 'prop-types';
import { Popover, Icon } from 'antd';
import CreateListFormModal from './CreateListFormModal';
import PopoverContent from './PopoverContent';

const CreateListAction = ({
  visiblePop,
  visibleMod,
  showPopover,
  showDialog,
  hideModal,
  myLists,
  movieId,
  addMovieToList,
}) => {
  return (
    <>
      <Popover
        content={
          <PopoverContent
            showDialog={showDialog}
            myLists={myLists}
            addMovieToList={addMovieToList}
            movieId={movieId}
          />
        }
        title="Add movie to list"
        visible={visiblePop}
        trigger="click"
        onVisibleChange={showPopover}
      >
        <Icon type="plus-circle" />
      </Popover>
      <CreateListFormModal visibleMod={visibleMod} hideModal={hideModal} />
    </>
  );
};

CreateListAction.propTypes = {
  visiblePop: PropTypes.bool.isRequired,
  visibleMod: PropTypes.bool.isRequired,
  showPopover: PropTypes.func.isRequired,
  showDialog: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
  myLists: PropTypes.arrayOf(PropTypes.object).isRequired,
  movieId: PropTypes.number.isRequired,
  addMovieToList: PropTypes.func.isRequired,
};

export default CreateListAction;
