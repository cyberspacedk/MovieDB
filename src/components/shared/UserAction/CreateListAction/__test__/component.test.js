import React from 'react';
import { shallow } from 'enzyme';
import CreateListAction from '../component';

describe('Component: CreateListAction', () => {
  const props = {
    visiblePop: false,
    visibleMod: false,
    showPopover: jest.fn(),
    showDialog: jest.fn(),
    hideModal: jest.fn(),
    addMovieToList: jest.fn(),
    myLists: [{}],
    movieId: 777,
  };

  const wrapper = shallow(<CreateListAction {...props} />);

  it('Match its snapshot. Item provided', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
