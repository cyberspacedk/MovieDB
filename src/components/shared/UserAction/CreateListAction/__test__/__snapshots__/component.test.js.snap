// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CreateListAction Match its snapshot. Item provided 1`] = `
<Fragment>
  <Popover
    content={
      <PopoverContent
        addMovieToList={[MockFunction]}
        movieId={777}
        myLists={
          Array [
            Object {},
          ]
        }
        showDialog={[MockFunction]}
      />
    }
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    onVisibleChange={[MockFunction]}
    overlayStyle={Object {}}
    placement="top"
    title="Add movie to list"
    transitionName="zoom-big"
    trigger="click"
    visible={false}
  >
    <Icon
      type="plus-circle"
    />
  </Popover>
  <Connect(WithFormik(CreateListFormModalContainer))
    hideModal={[MockFunction]}
    visibleMod={false}
  />
</Fragment>
`;
